src/js/components/JobForm.js

```diff
@@ -43,21 +37,60 @@ class JobForm extends SchemaForm {
       return;
     }
 
-    this.validateForm();
-    // Provide updated model to get new multipleDefinition
-    let model = this.triggerTabFormSubmit();
-    this.multipleDefinition = this.getNewDefinition(model);
-    this.props.onChange(this.getDataTriple(model));
-    // Only update if schedule changed
-    if (scheduleEnabledChange) {
+    // Fetch the updated model from the <Form /> component
+    // nested deeper in the component tree.
+    this.model = this.triggerTabFormSubmit();
```

Comment:
Yeah, I also try to avoid it as much as possible, but it comes fro the superclass `SchemaForm` that uses it. This is really nasty and the solution here is what I thought can be the minimal effort required to make this work without major refactoring on the `SchemaForm` (since there is already a plan for this).